src/main/java/nl/workingtalent/wtacademy/reservation/ReservationController.java

```diff
@@ -1,11 +1,140 @@
 package nl.workingtalent.wtacademy.reservation;
 
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import nl.workingtalent.wtacademy.book.Book;
+import nl.workingtalent.wtacademy.book.BookService;
+import nl.workingtalent.wtacademy.dto.ResponseDto;
+import nl.workingtalent.wtacademy.user.User;
+import nl.workingtalent.wtacademy.user.UserService;
+
 @RestController
+@CrossOrigin(maxAge = 3600)
 public class ReservationController {
 
 	@Autowired
 	private ReservationService service;
+
+	@Autowired
+	private BookService bookService;
+
+	@Autowired
+	private UserService userService;
+
+	// READ
+	@RequestMapping("reservation/all")
+	public ResponseDto findAllReservations() {
+		List<Reservation> reservations = service.findAllReservations();
+
+		if (reservations.isEmpty()) {
+			return new ResponseDto(false, null, null, "No reservations found.");
+		}
+
+		Stream<ReadReservationDto> readReservationDtoStream = getReservations(reservations);
+		return new ResponseDto(true, readReservationDtoStream, null,
+				reservations.size() + (reservations.size() < 2 ? " reservation " : " reservations ") + "found.");
+	}
+
+	@RequestMapping("reservation/{id}")
+	public ResponseDto findReservationById(@PathVariable("id") long id) {
+		Optional<Reservation> reservationOptional = service.findReservationById(id);
+
+		if (reservationOptional.isPresent()) {
+			Reservation reservation = reservationOptional.get();
+			ReadReservationDto readReservationDto = new ReadReservationDto(reservation);
+			return new ResponseDto(true, readReservationDto, null, "Reservation found.");
+		}
+		return new ResponseDto(false, null, null, "No reservation found.");
+	}
+
+	@RequestMapping("reservation/search")
+	public ResponseDto searchReservation(@RequestBody SearchReservationDto dto) {
+		List<Reservation> reservations = service.searchReservations(dto);
+		Stream<ReadReservationDto> dtos = reservations.stream().map((reservation) -> {
+			return new ReadReservationDto(reservation);
+		});
+		return new ResponseDto(true, dtos, null,
+				reservations.size() + (reservations.size() < 2 ? " reservation " : " reservations ") + "found.");
+	}
```

Comment:
Misschien dat je de mogelijkheid wil om te searchen tussen bepaalde datums. Of bijvoorbeeld alleen alle reservations van de laatste 10 dagen. Op dit moment niet het belangrijkst.

Reply:
Ik zal het in de backlog zetten, zodat we het niet vergeten.